packages/core/src/odata/common/request/odata-request-config.ts

```diff
@@ -14,24 +15,44 @@ export abstract class ODataRequestConfig {
   payload: Record<string, any> | string;
   customServicePath: string;
 
+  readonly defaultHeaders: Record<string, any> = {
+    'content-type': 'application/json',
+    accept: 'application/json'
+  };
+
   private _customHeaders: Record<string, string> = {};
   private _customQueryParameters: Record<string, string> = {};
 
+  /**
+   * @deprecated Since v1.30.0. Use [[defaultHeaders]] instead.
+   */
+  get contentType() {
+    return this.defaultHeaders['content-type'];
+  }
+
   /**
    * Creates an instance of ODataRequest.
    *
    * @param method - HTTP method of the request
    * @param defaultServicePath - default path of the according service
-   * @param contentType - The content type of the request
+   * @param defaultHeadersOrContentType - The default headers of the given request as an object. When passing a string only set the content type header will be set. Setting the content type only is deprecated since v1.30.0.
    */
   constructor(
     public method: RequestMethodType,
     readonly defaultServicePath: string,
-    readonly contentType = 'application/json'
+    defaultHeadersOrContentType?: Record<string, any> | string
```

Comment:
Could you overload the constructor and deprecate the one with the string argument. Then it is easier to find and remove later on.

Reply:
I did this initially but it is against our lint rules. I can turn them off, but I am not sure if it is possible to define multiple docs for this. I'll check.

Reply:
Forgot to add it, but will do on the next PR.